Tests/FluentAssertions.Shared.Specs/ObjectAssertionSpecs.cs

```diff
@@ -261,7 +261,83 @@ public void When_object_type_is_exactly_equal_to_the_specified_type_it_should_no
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () => valueTypeObject.Should().NotBeOfType(typeof(double));
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldNotThrow();
+        }
+
+        [TestMethod]
+        public void When_object_type_is_value_type_and_matches_received_type_not_as_expected_should_fail()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            int valueTypeObject = 42;
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () => valueTypeObject.Should().NotBeOfType(typeof(int));
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>();
+        }
+
+        [TestMethod]
+        public void When_object_type_is_value_type_and_doesnt_match_received_type_should_fail()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            int valueTypeObject = 42;
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () => valueTypeObject.Should().BeOfType(typeof(double));
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>();
```

Comment:
Please also include some checks on the message so it's clear that the expected and actual types are included.